rmw_zenoh_cpp/src/detail/rmw_publisher_data.cpp

```diff
@@ -246,18 +238,18 @@ rmw_ret_t PublisherData::publish(
   // session use different encoding formats. In our case, all key expressions
   // will be encoded with CDR so it does not really matter.
   zenoh::ZResult result;
-  auto options = zenoh::Publisher::PutOptions::create_default();
-  options.attachment = create_map_and_set_sequence_num(
-    sequence_number_++,
-    entity_->copy_gid());
+  int64_t source_timestamp = rmw_zenoh_cpp::get_system_time_in_ns();
+  auto opts = zenoh::ext::AdvancedPublisher::PutOptions::create_default();
+  opts.put_options.attachment = rmw_zenoh_cpp::AttachmentData(
+    sequence_number_++, source_timestamp, entity_->copy_gid()).serialize_to_zbytes();
```

Comment:
It looks like we're switching from `create_map_and_set_sequence_num()` to `AttachmentData` in this PR which was not part of #368.
Do we still need  `create_map_and_set_sequence_num`?

Reply:
I think it's best if we do a backport of https://github.com/ros2/rmw_zenoh/pull/294 but make modifications to drop the `tracetools` dependency. That should ensure we're consistent with our usage of `AttachmentData` instead of `create_map_and_set_sequence_num`. Please hold off on merging this and the humble backport in the meanwhile.